nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,83 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+//--------------------------------------------------
+//Respond with the json for the meal with the corresponding id
+router.get("/:id", async (request, response) => {
+  if (isNaN(request.params.id)) {
+    return response.status(400).send("Bad Request: Uncorrect id type");
+  }
+  const mealsWithId = meals.filter(
+    (meal) => meal.id === parseInt(request.params.id)
+  );
+  /*
+  if (mealsWithId.length === 0) {
+    return response.status(404).send("Not Found: no meal for this request");
+  }*/
+
+  return response.status(200).json(mealsWithId[0]);
+  //console.log("in /api/meals");
+});
+
+//--------------------------------------------------
+router.get("/", async (request, response) => {
+  //What if the users writes a query parameter that is not supported?
+  ////////////////////////////////////
```

Comment:
Good point. In general the server must validate if the incoming parameters (in query, path or body) are valid. For this you can create a validation function. For example you can validate if the maxPrice is a number. If it is not, we should throw an error (return an error status). 
Similarly, if the query parameter is not maxPrice, limit, createdAfter, etc, you can also throw an error (return an error status).